### PR TITLE
Recipe-App Capstone Project

### PR DESCRIPTION
**Pull request by omarsalem7**
## What does this PR do?

This PR is raised to recipe app capstone project

## Description of the task to be completed

- [x] Login page and registration page:
    - Should be built with Devise.
- [x] Food list:
    - Should display a list of food as in the wireframe.
    - Should lead to the form that allows users to add new food.
- [x] Recipies list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow user to delete it.
- [x] Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- [x] Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows user to add new food.
- [x] General shopping list view :
    - Should show the list of food that is missing for all recipes of the logged-in user.
    - Should count the total food items and total price of the missing food.
- [x] Create a navigation menu that allows users to open all of the pages you created.
  
## How should it be tested

Clone the repository
```
git clone https://github.com/omarsalem7/recipe-app.git
```
Navigate to the folder
```
cd recipe-app
```
Checkout the branch
```
git checkout dev
```
## to run tests
```
rspec ./spec/models
rspec ./spec/features
```

#### Thanks for reviewing our PR 😊

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          <tr>
+            <th scope="col">Food</th>
+            <th scope="col">Quantity</th>
+            <th scope="col">Price</th>
+          </tr>
+        </thead>
+        <tbody>
+          <%= render "shopping_lists/shopping_list" %>
+        </tbody>
+      </table>
```
Comment (ShoiraTa) [inline app/views/recipes/show.html.erb:55]: - [ ] Kindly add Action buttons, modify / remove in ingredients  you can set them as `disabled` 

Yous  | Wireframe
------------- | -------------
Screenshot 2022-05-27 at 18 33 06  | Screenshot 2022-05-27 at 18 31 53

File: app/views/recipes/public.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<h1 class="d-flex justify-content-center">Public recipes</h1>
+<div>
+  <% @public_recipes.each do |recipe| %>
+    <div class="card mb-3">
+      <div class="card-body">
```
Comment (ShoiraTa) [inline app/views/recipes/public.html.erb:5]: - [ ] Clicking on a recipe item in Public recipes page should lead to recipe details. But in your application I, on public page, cannot click on the recipe .

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi team,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Using only necessary routes ✔️ 
- Using capybara to test ✔️ 
- Application works as expected ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omarsalem7** [comment]
Hello @ShoiraTa 😊
we did the changes, you can check them now

**Comment by ShoiraTa** [comment]
## **STATUS: APPROVED  🟢 ** 
------
### Hi team, 

**Great work on making the changes 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**